core/src/main/scala/org/apache/spark/scheduler/cluster/SparkDeploySchedulerBackend.scala

```diff
@@ -31,16 +34,16 @@ private[spark] class SparkDeploySchedulerBackend(
   with AppClientListener
   with Logging {
 
-  var client: AppClient = null
-  var stopping = false
-  var shutdownCallback : (SparkDeploySchedulerBackend) => Unit = _
-  @volatile var appId: String = _
+  private var client: AppClient = null
+  private var stopping = false
+  private val shutdownCallback: AtomicReference[(SparkDeploySchedulerBackend) => Unit] =
```

Comment:
@sarutak  would this not be simpler as a `@volatile` reference? that's how `appId` is handled below.